src/shopping-list/shopping-list.js

```diff
@@ -0,0 +1,98 @@
+import { getIngredients, getRecipes } from "./shopping-list-api.js";
+
+const searchInput = document.querySelector('.recipe-search__input--js');
+const displayedRecipiesList = document.querySelector('.list-recipies__container--js');
+const shoppingListContainer = document.querySelector('.shopping-list__ingredients-container');
+const shoppingListParagraph = document.querySelector('.shopping-list__paragraph');
+
+const shoppingList = [];
+
+let delayTimer;
+
+searchInput.addEventListener('keyup', async () => {
+  clearTimeout(delayTimer);
+  delayTimer = setTimeout(async function() {
```

Comment:
I think you should clean settimeout to avoid memory leaks

Reply:
I'm not sure if I understand this comment, please kindly elaborate if necessary